airflow/providers/google/cloud/hooks/bigquery.py

```diff
@@ -67,7 +67,9 @@
 class BigQueryHook(GoogleBaseHook, DbApiHook):
     """Interact with BigQuery. This hook uses the Google Cloud connection."""
 
-    conn_name_attr = 'gcp_conn_id'  # type: str
+    conn_name_attr = 'gcp_conn_id'
+    default_conn_name = 'google_cloud_default'
+    conn_type = 'google_cloud_platform'
```

Comment:
Shouldn't this be something like `conn_type = 'big_query'`?

Reply:
Possibly. I just do not want to fix all those in this PR. I simply converted what I found in the original connection.py. 

I propose that we start agreeing the approach there and standardize those, but this shoudl be a completely different PR (and if you want to propose some guidelines and standard/consistent approach for that - feel free). 

I do not want to scope-creep that change.

Reply:
We don't need a separate type for BigQuery. In Airflow 1.10, we had a few more connection types, but in Airflow 2.0, this was standardized and most connections use one connection type. In some exceptional situations we need additional configuration options and then we use a new type of connection.
See: http://apache-airflow-docs.s3-website.eu-central-1.amazonaws.com/docs/apache-airflow-providers-google/latest/connections/index.html
https://github.com/apache/airflow/blob/master/UPDATING.md#normalize-gcp_conn_id-for-google-cloud

Reply:
@potiuk :+1: 

@mik-laj If we want to be able to get a BigQuery hook to from `Connection.get_hook` we'll need a unique type won't we?

(Right now the conn type for BQ is google_cloud_platform, but that means, say, DataProc can't be google_cloud_platform)

Reply:
@ashb Since BigQuery implements DBApiHook, we should probably create a new connection type to be used by DbApiHook.

I am afraid of registering all hooks because that will also mean that they will be loaded. And loading all modules from all Google libraries into memory is an expensive operation.

Reply:
Oh good to know.

We'll have to come up with some approach to registereing the information for UI without importing the hook classes then.

Reply:
Related code: 
https://github.com/apache/airflow/blob/a6cf883513280de2df80989b81eb65065ec761d1/airflow/providers_manager.py#L199-L209